lib/ansible/modules/system/mount.py

```diff
@@ -521,14 +521,8 @@ def get_linux_mounts(module, mntinfo_file="/proc/self/mountinfo"):
     """Gather mount information"""
 
     try:
-        f = open(mntinfo_file)
-    except IOError:
-        return
-
-    lines = map(str.strip, f.readlines())
-
-    try:
-        f.close()
+        with open(mntinfo_file) as f:
+            lines = map(str.strip, f.readlines())
```

Comment:
You are changing semantics here. Originally, errors while opening the file were ignored, the function simply returned. Errors while reading the file's content were escalated, and errors while closing the file were handled (by failing the module).

Now, you're handling **all** errors by failing the module. You should definitely **not** handle file not found errors this way! (No idea about other errors while opening the file.)

Reply:
Yes. I don't really see how it could be handled differently, in a nice way. There are some error codes that may be useful, https://docs.python.org/2/library/errno.html The exception for opening the file can be caused by different events. Can we consider a compromise to simplify this so that the `IOError` is used for the different events? I think that's also cleaner and more maintainable. Python 3 has `FileNotFoundError` but it's not available in Python 2.